local-volume/provisioner/pkg/populator/populator.go

```diff
@@ -41,8 +39,8 @@ func NewPopulator(config *common.RuntimeConfig) *Populator {
 	return &Populator{RuntimeConfig: config}
 }
 
-// Start launches the PV informer
-func (p *Populator) Start() {
+// Start launches the PV informer and waits for cache to sync, returns error if it fails.
+func (p *Populator) Start(stopCh <-chan struct{}) error {
 	_, controller := kcache.NewInformer(
```

Comment:
Would using SharedInformers be better here for both PV and StorageClass objects?

Reply:
No, IMO [`SharedInformer`](https://github.com/kubernetes/client-go/blob/v7.0.0/tools/cache/shared_informer.go#L34) is preferred if there are multiple listeners for same cache. For examples, two controllers who both will listen on events of PersistentVolume can use `SharedInformer ` to share data cache. In local volume provisioner, PersistentVolums cache is only consumed Populator,  StorageClasse Cache only by Discoverer.

Reply:
The interface seems simpler as you don't have to start separate controllers

Reply:
Agree. I'll updated to use `SharedInformer`.

Reply:
Updated in https://github.com/kubernetes-incubator/external-storage/pull/776/commits/af8c713a0e7c3bf30eeada9aef449f0baf7cda6a.

I've done some more refactoring which I found necessary:

- Use `informers.SharedInformerFactory` to create SharedInformers for PersistentVolumes, StorageClasses, and Jobs.
  - Note that, for Jobs we are using a customized constructor
- With `informers.SharedInformerFactory`, it's easier to start informers and wait for all informers' cache were synced before running controller logic
  - Previously, `JobController` waits for cache in a separate go routine which is not right, because it is not a standalone controller from my perspective.